### ISSUE TITLE
Table-data Browser: On the variants table, Something seems amiss when there are 692771 rows, and the user is given a mechanism to page through those records, 250 records per page, which means there are about 2771 pages.

### ISSUE THREAD

**Issue by leehart** [wtchg-kwiatkowski/observatory-web#242]
(Maybe allow the user to type in a page number, or row number.)

**Comment by benjeffery** [comment]
What is amiss? 692771/250 = 2771 right?

**Comment by leehart** [comment]
Sure, there's nothing wrong with that calculation. (The webapp doesn't tell you how many pages there are, that was just my own calculation, which is why I said "about", because I can't be confident!) There's something wrong with expecting a user to page through 2771 pages, one-by-one, amiright!? Imagine you wanted to see what was on page 1001. I don't know what world that would be, but it would be the same world where you had to click that little next-page icon _about_ 1000 times! :1st_place_medal: 

As per the initial comment, I reckon something like a page icon, nestled between the previous-page and next-page icon-buttons, which when pressed allows the user to enter a page number, even if it's an old-school input dialog, with some validation, should patch up this nicely.
I suppose that solution might also necessitate showing the current and maximum page numbers, though.

**Comment by benjeffery** [comment]
But how did the user know what they wanted was on 1001 without flicking though 1-1001?

**Comment by leehart** [comment]
The answer is on page 1001 of the variants table. See for yourself!

**Comment by benjeffery** [comment]
Ok, send me the link.

I just don't see the use case - if you are trying to find something you make a filter to find it.
Google has a control to let you see 10 pages forward. Why would you want to go deeper?

**Comment by leehart** [comment]
You're right, I can't point to a genuine use-case either, which is why I've had to resort to vague hand-waving and absurdly extreme hypotheticals, in the hope of congeniality.

I reckon there are a couple of important differences with Google's scenario:
1. the default user-interface is a simple text-search box; if your search term didn't give you instant satisfaction, then you have to reluctantly change it (it seems like your fault)
2. there are potentially endless results of undetermined character

Whereas, in obs-web:
1. the default user-interface is a table of information that we've prepared; if the user can't find what they're looking with instant satisfaction, then they have to reluctantly create a filter, which is a pain, and that seems like our fault (not theirs)
2. there is a finite number of records of known structure

But I think there is one important similarity:
- if you're too stubborn to change your search term / change the default query, then you can progress to page 11, 12, 13, etc., easily enough

If there is a genuine use-case in the wild, I expect it would probably take a very rare and specific form, something like:
1. Héloïse is looking for variants with certain properties
2. Héloïse realises that there are 692771 rows
3. Héloïse realises that the first and last page features won't help them find the variants
4. Héloïse realises that sorting the various columns in different ways doesn't help
5. Héloïse realises that they need to add a filter to the table
6. Héloïse tries to add a filter, but the interface is confusing or their query is not simple, arrives at a partial solution
7. Héloïse realises that there are still about 6000 variants to search through
8. Héloïse realises that the variants that they're looking for are probably on page 12
9. Héloïse tries to remove columns and sort columns, but it doesn't help
10. Héloïse reluctantly clicks the next page button a dozen times, wondering why there isn't a "go to page" feature.


tl:dr
I'm happy to see this so-called issue buried, until we receive Héloïse's email!

...but I still think a little page icon nestled in between those previous and next icons would look cool, even if it's never used, and it would reinforce the meaning of those arrow icons.
Purely use-case-driven development has its limitations, imo. I wonder what proportion of this webapp is based on whim! Are the Users even real, Tron?!

Here's that link: https://xkcd.com/1309/ ;-)
